flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/jobmanager/JobManagerFlinkMemoryUtils.java

```diff
@@ -88,16 +88,16 @@ private static void verifyJvmHeapSize(MemorySize jvmHeapSize) {
 		}
 	}
 
-	private static void verifyJobStoreCacheSize(Configuration config, MemorySize jvmHeapSize) {
+	private static void verifyJobStoreCacheSize(Configuration config, MemorySize offHeapMemorySize) {
```

Comment:
This method is correct as the job cache is part of the JVM heap, not off-heap.
The problem is that `verifyJobStoreCacheSize` is called with `offHeapMemory` in `JobManagerFlinkMemoryUtils#createJobManagerFlinkMemory`:
`verifyJobStoreCacheSize(config, offHeapMemory);`
which should be:
`verifyJobStoreCacheSize(config, jvmHeap);` there.

Reply:
> This method is correct as the job cache is part of the JVM heap, not off-heap.
> The problem is that `verifyJobStoreCacheSize` is called with `offHeapMemory` in `JobManagerFlinkMemoryUtils#createJobManagerFlinkMemory`:
> `verifyJobStoreCacheSize(config, offHeapMemory);`
> which should be:
> `verifyJobStoreCacheSize(config, jvmHeap);` there.

@azagrebin Sorry for previous misunderstanding for the problem. And I have updated this method.